instrumentation/message-handler/library/src/main/java/io/opentelemetry/instrumentation/messagehandler/BatchMessageHandler.java

```diff
@@ -0,0 +1,172 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.instrumentation.messagehandler;
+
+import io.opentelemetry.api.OpenTelemetry;
+import io.opentelemetry.api.trace.Span;
+import io.opentelemetry.api.trace.SpanBuilder;
+import io.opentelemetry.api.trace.SpanContext;
+import io.opentelemetry.api.trace.SpanKind;
+import io.opentelemetry.api.trace.TracerBuilder;
+import io.opentelemetry.api.trace.propagation.W3CTraceContextPropagator;
+import io.opentelemetry.context.Context;
+import io.opentelemetry.context.Scope;
+import io.opentelemetry.context.propagation.TextMapGetter;
+import io.opentelemetry.contrib.awsxray.propagator.AwsXrayPropagator;
+import io.opentelemetry.instrumentation.api.instrumenter.messaging.MessageOperation;
+import io.opentelemetry.semconv.trace.attributes.SemanticAttributes;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.Locale;
+import java.util.Map;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+public abstract class BatchMessageHandler<T> {
```

Comment:
The design of this class seems to be very much geared towards X-Ray. The OpenTelemetry propagator API is flexible enough that this should not be necessary at all. E.g. even a fallback from X-Ray to a secondary propagator could be implemented as a composite propagator.

IMHO, the class should be refactored so that all X-Ray-specific code can be moved to SqsBatchMessageHandler.

Also, it should not rely on the assumption that only W3C and X-Ray exist, but work with any propagator

Reply:
@Oberon00 
Thanks for the feedback. Please re-review.